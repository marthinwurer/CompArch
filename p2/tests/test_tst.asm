;
; tests TST
;
; uses instructions MOV, BPT, HALT
;
; uses addressing modes 0, 6
;
	.globl	start
;
v1:	.word	0
v2:	.word	177777
v3:	.word	077777
;
start:	mov	v1,r1
	mov	v2,r2
	mov	v3,r3
	bpt
;
	tst	r1
	tst	r2
	tst	r3
;
	tst	v1
	tst	v2
	tst	v3
;
	halt
;
	.end	start
