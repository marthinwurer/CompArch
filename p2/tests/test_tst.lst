      1                                 ;
      2                                 ; tests TST
      3                                 ;
      4                                 ; uses instructions MOV, BPT, HALT
      5                                 ;
      6                                 ; uses addressing modes 0, 6
      7                                 ;
      8 000000                                   .globl  start
      9                                 ;
     10 000000  000000                  v1:      .word   0
     11 000002  177777                  v2:      .word   177777
     12 000004  077777                  v3:      .word   077777
     13                                 ;
     14 000006  016701  177766          start:   mov     v1,r1
     15 000012  016702  177764                   mov     v2,r2
     16 000016  016703  177762                   mov     v3,r3
     17 000022  000003                           bpt
     18                                 ;
     19 000024  005701                           tst     r1
     20 000026  005702                           tst     r2
     21 000030  005703                           tst     r3
     22                                 ;
     23 000032  005767  177742                   tst     v1
     24 000036  005767  177740                   tst     v2
     25 000042  005767  177736                   tst     v3
     26                                 ;
     27 000046  000000                           halt
     28                                 ;
     29 000050  000006'                          .end    start


pc     = 000007 %  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000006 RG 
v1       000000 R  
v2       000002 R  
v3       000004 R  
