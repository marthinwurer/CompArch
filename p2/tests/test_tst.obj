000 2 00 00
002 2 ff ff
004 2 ff 7f
006 4 c1 1d f6 ff
00a 4 c2 1d f4 ff
00e 4 c3 1d f2 ff
012 2 03 00
014 2 c1 0b
016 2 c2 0b
018 2 c3 0b
01a 4 f7 0b e2 ff
01e 4 f7 0b e0 ff
022 4 f7 0b de ff
026 2 00 00
006
