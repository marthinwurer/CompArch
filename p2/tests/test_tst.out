CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 6
000006:  PS[00]  016701 177766  MOV 177766(R7),R1
000012:  PS[04]  016702 177764  MOV 177764(R7),R2
000016:  PS[10]  016703 177762  MOV 177762(R7),R3
000022:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[177777]  R3[077777]
  R4[000000]  R5[000000]  R6[000000]  R7[000024]
000024:  PS[00]  005701  TST R1
000026:  PS[04]  005702  TST R2
000030:  PS[10]  005703  TST R3
000032:  PS[00]  005767 177742  TST 177742(R7)
000036:  PS[04]  005767 177740  TST 177740(R7)
000042:  PS[10]  005767 177736  TST 177736(R7)
000046:  PS[00]  000000  HALT

Machine Halted - HALT instruction

Simulated time 81 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
